src/lib/UniversalDiscovery/Event/Subscriber/RichTextEmbedAllowedContentTypes.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\UniversalDiscovery\Event\Subscriber;
+
+use eZ\Publish\API\Repository\ContentTypeService;
+use eZ\Publish\API\Repository\PermissionResolver;
+use eZ\Publish\API\Repository\Values\ContentType\ContentType;
+use eZ\Publish\API\Repository\Values\User\Limitation\ContentTypeLimitation;
+use EzSystems\EzPlatformAdminUi\Permission\PermissionCheckerInterface;
+use EzSystems\EzPlatformAdminUi\UniversalDiscovery\Event\ConfigResolveEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class RichTextEmbedAllowedContentTypes implements EventSubscriberInterface
+{
+    /** @var string[] */
+    private $restrictedContentTypesIdentifiers;
+
+    /**
+     * @param \eZ\Publish\API\Repository\PermissionResolver $permissionResolver
+     * @param \EzSystems\EzPlatformAdminUi\Permission\PermissionCheckerInterface $permissionChecker
+     * @param \eZ\Publish\API\Repository\ContentTypeService $contentTypeService
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    public function __construct(
+        PermissionResolver $permissionResolver,
+        PermissionCheckerInterface $permissionChecker,
+        ContentTypeService $contentTypeService
+    ) {
+        $this->restrictedContentTypesIdentifiers = $this->getRestrictedContentTypesIdentifiers(
+            $permissionResolver,
+            $permissionChecker,
+            $contentTypeService
+        );
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\PermissionResolver $permissionResolver
+     * @param \EzSystems\EzPlatformAdminUi\Permission\PermissionCheckerInterface $permissionChecker
+     * @param \eZ\Publish\API\Repository\ContentTypeService $contentTypeService
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    private function getRestrictedContentTypesIdentifiers(
+        PermissionResolver $permissionResolver,
+        PermissionCheckerInterface $permissionChecker,
+        ContentTypeService $contentTypeService
+    ): array {
+        $access = $permissionResolver->hasAccess('content', 'read');
+        if (!\is_array($access)) {
+            return [];
+        }
+
+        $restrictedContentTypesIds = $permissionChecker->getRestrictions($access, ContentTypeLimitation::class);
+
+        if (empty($restrictedContentTypesIds)) {
+            return [];
+        }
+
+        $restrictedContentTypes = $contentTypeService->loadContentTypeList($restrictedContentTypesIds);
+
+        return array_values(array_map(function (ContentType $contentType): string {
```

Comment:
I don't see any reason not to:
```suggestion
        return array_values(array_map(static function (ContentType $contentType): string {
```

Reply:
Done in https://github.com/ezsystems/ezplatform-admin-ui/pull/1441/commits/0e3c6cf95726dd0e52c9303cd701b00952299a51.